src/index.js

```diff
@@ -219,33 +225,44 @@ class Mongo {
     return profile;
   }
 
-  async createSession(userId: string, ip: string, userAgent: string): Promise<string> {
-    const ret = await this.sessionCollection.insertOne({
+  async createSession(userId: string, ip: string, userAgent: string,
+    extraData: ?string): Promise<string> {
+    let session = {
       userId,
       userAgent,
       ip,
+      extraData,
       valid: true,
-      [this.options.timestamps.createdAt]: Date.now(),
-      [this.options.timestamps.updatedAt]: Date.now(),
-    });
+      [this.options.timestamps.createdAt]: this.options.dateProvider(),
+      [this.options.timestamps.updatedAt]: this.options.dateProvider(),
+    };
+    if (this.options.idProvider) {
+      session = {
+        ...session,
+        _id: this.options.idProvider(),
+      };
+    }
+    const ret = await this.sessionCollection.insertOne(session);
     return ret.ops[0]._id;
   }
 
   async updateSession(sessionId: string, ip: string, userAgent: string): Promise<void> {
-    await this.sessionCollection.update({ _id: toMongoID(sessionId) }, {
+    const id = this.options.convertSessionIdToMongoObjectId ? toMongoID(sessionId) : sessionId;
+    await this.sessionCollection.update({ _id: id }, {
```

Comment:
instead of naming with `id` please name the const as `_id` and then you can change the selector to `{ _id }`